src/core/core-varlink.c

```diff
@@ -85,14 +87,19 @@ static int build_managed_oom_json_array_element(Unit *u, const char *property, s
         else if (streq(property, "ManagedOOMMemoryPressure")) {
                 mode = managed_oom_mode_to_string(c->moom_mem_pressure);
                 use_limit = true;
+                limit = c->moom_mem_pressure_limit;
+        } else if (streq(property, "ManagedOOMMemoryPressureDurationUSec")) {
+                mode = managed_oom_mode_to_string(c->moom_mem_pressure);
+                use_limit = true;
+                limit = c->moom_mem_pressure_duration_usec;
         } else
                 return -EINVAL;
 
         return sd_json_buildo(ret_v,
                               SD_JSON_BUILD_PAIR("mode", SD_JSON_BUILD_STRING(mode)),
                               SD_JSON_BUILD_PAIR("path", SD_JSON_BUILD_STRING(crt->cgroup_path)),
                               SD_JSON_BUILD_PAIR("property", SD_JSON_BUILD_STRING(property)),
-                              SD_JSON_BUILD_PAIR_CONDITION(use_limit, "limit", SD_JSON_BUILD_UNSIGNED(c->moom_mem_pressure_limit)));
+                              SD_JSON_BUILD_PAIR_CONDITION(use_limit, "limit", SD_JSON_BUILD_UNSIGNED(limit)));
```

Comment:
Oh good I'm not the only one confused by this :)

So right now, the message dbus sends to systemd-oomd for a cgroup path is an array like:
```
[
  {
    "mode": "kill",
    "path": "/blah.slice",
    "property": "ManagedOOMSwap"
  },
  {
    "mode": "kill",
    "path": "/blah.slice",
    "property": "ManagedOOMMemoryPressure",
    "limit": 100
  },
  {
    "mode": "kill",
    "path": "/blah.slice",
    "property": "ManagedOOMMemoryPressureDurationUSec",
    "limit": 200
  },
]
```

Then the parsing is equally weird...it inserts into systemd-oomd cgroup path -> context hash map for the first message, then overwrites limit and duration on successive messages: https://github.com/systemd/systemd/blob/5b80cef69f00052977ae51d27f21c631266a57ec/src/oom/oomd-manager.c#L57

I did the "clever" (read: hacky) thing and I just added another property to the array.

Since you're also confused by this, I'm gonna change the message format to a single dict and not an array:
```
  {
    "mode": "kill",
    "path": "/blah.slice",
    "type": SWAP, # or MEMORY_PRESSURE enum
    "limit": 100,
    "duration": 200
  }
```